src/v/cloud_storage/async_manifest_view.h

```diff
@@ -203,4 +221,268 @@ class materialized_manifest_cache {
     ss::semaphore_units<> _reserved;
 };
 
-} // namespace cloud_storage
+class async_manifest_view;
+
+/// The structure represents evicted manifest. The difference between this
+/// state and uninitialized state is that 'stale_manifest' will trigger
+/// timed_out error on first access.
+struct stale_manifest {
+    /// Cached last offset value + 1 of the stale manifest
+    model::offset next_offset;
+};
+
+/// Type that represents section of full metadata. It can either contain
+/// a materialized spillover manifest or reference to main STM manifest.
+using manifest_section_t = std::variant<
+  std::monostate,
+  stale_manifest,
+  ss::shared_ptr<materialized_manifest>,
+  std::reference_wrapper<const partition_manifest>>;
+
+/// Result of the ListObjectsV2 scan
+struct spillover_manifest_list {
+    /// List of manifest paths
+    std::deque<remote_manifest_path> manifests;
+    /// List of decoded path components (offsets and timestamps)
+    std::deque<spillover_manifest_path_components> components;
+    /// List of manifest sizes (in binary format)
+    std::deque<size_t> sizes;
+};
+
+class async_manifest_view_cursor;
+
+/// Service that maintains a view of the entire
+/// partition metadata including STM manifest and
+/// all spillover manifests. Spillover manifests are
+/// materialized on demand.
+class async_manifest_view {
+    friend class async_manifest_view_cursor;
+
+public:
+    async_manifest_view(
+      ss::sharded<remote>& remote,
+      ss::sharded<cache>& cache,
+      const partition_manifest& stm_manifest,
+      cloud_storage_clients::bucket_name bucket,
+      partition_probe& probe);
+
+    ss::future<> start();
+    ss::future<> stop();
+
+    /// Get active spillover manifests asynchronously
+    ///
+    /// \note the method may hydrate manifests in the cache or
+    ///       wait until the memory for the manifest becomes
+    ///       available.
+    ss::future<
+      result<std::unique_ptr<async_manifest_view_cursor>, error_outcome>>
+    get_active(async_view_search_query_t q) noexcept;
```

Comment:
nit: maybe rename `get_cursor()` or somesuch? With inferred types it's not always obvious what "active" will refer to